contrib/win32/win32compat/w32log.c

```diff
@@ -115,16 +118,36 @@ openlog_file()
 		wcscat_s(ssh_cfg_path, _countof(ssh_cfg_path), L"\\ssh"); /* "%programData%\\ssh" */
 
 		if ((wcsncat_s(log_file, PATH_MAX + 12, ssh_cfg_path, wcslen(ssh_cfg_path)) != 0) ||
-		    (wcsncat_s(log_file, PATH_MAX + 12, logs_dir, 6) != 0) ||
-		    (wcsncat_s(log_file, PATH_MAX + 12, tail + 1, wcslen(tail + 1) - 3) != 0 ) ||
-		    (wcsncat_s(log_file, PATH_MAX + 12, L"log", 3) != 0))
+			(wcsncat_s(log_file, PATH_MAX + 12, logs_dir, 6) != 0))
 			return;
+		
+		if (strcmp(identity, "sftp-server") == 0) {
+			wchar_t* id = utf8_to_utf16(identity);
+			if ((wcsncat_s(log_file, PATH_MAX + 12, id, wcslen(id)) != 0) ||
+				(wcsncat_s(log_file, PATH_MAX + 12, L".log", 4) != 0)) {
+				free(id);
+				return;
+			}
+			free(id);
+		}
+
+		if (strcmp(identity, "sshd") == 0)
+			if ((wcsncat_s(log_file, PATH_MAX + 12, tail + 1, wcslen(tail + 1) - 3) != 0) ||
+				(wcsncat_s(log_file, PATH_MAX + 12, L"log", 3) != 0))
+				return;
+
 	}
 	
-	errno_t err = _wsopen_s(&logfd, log_file, O_WRONLY | O_CREAT | O_APPEND, SH_DENYNO, S_IREAD | S_IWRITE);
-		
-	if (logfd != -1)
-		SetHandleInformation((HANDLE)_get_osfhandle(logfd), HANDLE_FLAG_INHERIT, 0);
+	errno_t err;
+	if (strcmp(identity, "sftp-server") == 0) {
+		err = _wsopen_s(&sftp_server_logfd, log_file, O_WRONLY | O_CREAT | O_APPEND, SH_DENYNO, S_IREAD | S_IWRITE);
+		if (sftp_server_logfd != -1)
+			SetHandleInformation((HANDLE)_get_osfhandle(sftp_server_logfd), HANDLE_FLAG_INHERIT, 0);
+	} else {
+		err = _wsopen_s(&logfd, log_file, O_WRONLY | O_CREAT | O_APPEND, SH_DENYNO, S_IREAD | S_IWRITE);
+		if (logfd != -1)
```

Comment:
I doubt if this is required. Child sshd process is not doing logging anymore.
remove this logic and compare the sshd.log file with and without this change to make sure we are not missing any child process.